src/uct/cuda/base/cuda_iface.c

```diff
@@ -39,8 +39,13 @@ uct_cuda_base_query_devices_common(
     ucs_sys_device_t sys_device = UCS_SYS_DEVICE_ID_UNKNOWN;
     CUdevice cuda_device;
 
-    if (cuCtxGetDevice(&cuda_device) == CUDA_SUCCESS) {
+    if (uct_cuda_base_is_context_active()) {
+        if (UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetDevice(&cuda_device)) != UCS_OK) {
+            return UCS_ERR_NO_DEVICE;
+        }
         uct_cuda_base_get_sys_dev(cuda_device, &sys_device);
+    } else {
+        ucs_debug("querying devices without active context");
```

Comment:
this means we won't know the sys device of the GPU? it can affect protocols and devices selection..

Reply:
Yes. And this is current behavior. `cuCtxGetDevice` returns `CUDA_ERROR_INVALID_CONTEXT`, if there is no active device.
We can try to add here `cuDeviceGet(&cuda_device, 0)`, assuming that 0-th device will be used...

Reply:
Yes, let's do this, and also update the debug message to print that no device is selected and we're taking locality information from device 0

Reply:
Reverted this change because of wire incompatibility issue. https://github.com/openucx/ucx/pull/9529